src/sentry/auth/providers/oauth2.py

```diff
@@ -53,15 +52,14 @@ def dispatch(self, request: Request, helper) -> Response:
 
         state = uuid4().hex
 
-        url_prefix = generate_url_prefix(request)
-        params = self.get_authorize_params(
-            state=state, redirect_uri=helper.get_redirect_url(url_prefix)
-        )
-        redirect_uri = f"{self.get_authorize_url()}?{urlencode(params)}"
+        params = self.get_authorize_params(state=state, redirect_uri=helper.get_redirect_url())
+        authorization_url = f"{self.get_authorize_url()}?{urlencode(params)}"
 
         helper.bind_state("state", state)
+        if request.subdomain:
+            helper.bind_state("subdomain", request.subdomain)
```

Comment:
Should a user start and SSO flow, leave the tab open with oauth dance incomplete and then open a second tab, not complete that sso, and then go back and complete the first SSO flow they *could* end up at the wrong place.

In practice that is unlikely to matter as they should have access to both orgs that they started SSO on.

Reply:
@markstory Wouldn't the first SSO fail if the `state` attributed was regenerated on the second tab? We compare the state values here when the user comes back to Sentry from the authentication provider https://github.com/getsentry/sentry/blob/8d9f92fafc7583a0279284690409b6fa9c420e5b/src/sentry/auth/providers/oauth2.py#L105-L106

Reply:
I do agree in practice it shouldn't matter.